src/components/common/facebook/FacebookWrapper.js

```diff
@@ -1,42 +1,22 @@
-import React, { Component } from 'react';
+import React, { useContext, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import withFB from './withFB';
+import FacebookContext from 'contexts/FacebookContext';
 
-class FacebookWrapper extends Component {
-  constructor(props) {
-    super(props);
-    this.container = null;
-    this.handleContainer = this.handleContainer.bind(this);
-  }
+const FacebookWrapper = ({ children }) => {
+  const FB = useContext(FacebookContext);
+  const container = useRef(null);
 
-  componentDidMount() {
-    if (this.props.FB && this.container) {
-      this.props.FB.XFBML.parse(this.container);
+  useEffect(() => {
+    if (FB && container.current) {
+      FB.XFBML.parse(container.current);
     }
-  }
+  });
```

Comment:
didMount 是不是dep應該要加上 `[]`

```suggestion
  }, []);
```

Reply:
componentDidUpdate 其實是每次都會觸發，這次也想讓 FB 針對 ref 指到的 dom 更新